src/app/static/src/tests/components/runWorkflow/runWorkflowProgress.test.ts

```diff
@@ -1,15 +1,88 @@
 import {shallowMount} from "@vue/test-utils";
 import runWorkflowProgress from '../../../js/components/runWorkflow/runWorkflowProgress.vue'
+import {mockAxios} from "../../mockAxios";
+import Vue from "vue";
+
+const workflows = {
+    "status": "success",
+    "errors": null,
+    "data": [
+        {date: "time1",
+        key: "key1",
+        name: "name1",
+        email: "email1"},
+        {date: "time2",
+        key: "key2",
+        name: "name2",
+        email: "email2"}
+    ]
+  }
+
+const workflowStatus1 = {
+    "status": "success",
+    "errors": null,
+    "data": {
+      "status": "running",
+      "reports": [
+        {
+          "key": "preterrestrial_andeancockoftherock",
+          "name": "report one a",
+          "status": "error",
+          "date": "2021-06-16T09:51:16Z"
+        },
+        {
+          "key": "hygienic_mammoth",
+          "name": "report two a",
+          "status": "success",
+          "version": "20210510-100458-8f1a9624",
+          "date": "2021-06-16T09:51:16Z"
+        },
+        {
+          "key": "blue_bird",
+          "name": "report three a",
+          "status": "running",
+          "date": null
+        }
+      ]
+    }
+  }
+
 
 describe(`runWorkflowProgress`, () => {
+    beforeEach(() => {
+        mockAxios.reset();
+        mockAxios.onGet('http://app/workflows')
+            .reply(200, workflows);
+        mockAxios.onGet('http://app/workflows/key1/status')
+            .reply(200, workflowStatus1);
+    });
 
     const getWrapper = () => {
         return shallowMount(runWorkflowProgress, {propsData: {workflowMetadata: {}}})
     }
 
-    it(`it can render runWorkflowProgress page`, () => {
+    it(`it can render if no workflows returned`, async (done) => {
+        mockAxios.onGet('http://app/workflows')
+            .reply(200, null);
+        const wrapper = getWrapper()
+        
+        setTimeout(() => {
+            expect(wrapper.find("p").text()).toBe("No workflows to show")
+            done();
+        })
+    })
+
+    it(`it can render runWorkflowProgress page`, async (done) => {
         const wrapper = getWrapper()
-        expect(wrapper.find("p").text()).toBe("Run workflow progress is coming soon")
+        
+        setTimeout(() => {
+            expect(wrapper.find("label").text()).toBe("Workflow")
+            expect(wrapper.find("v-select-stub").attributes("placeholder")).toBe("Select workflow or search by name...")
+            // Tests to be reinstated once buttons are integrated with workflow wizard
+            // expect(wrapper.findAll("button").at(0).text()).toBe("Clone workflow")
+            // expect(wrapper.findAll("button").at(1).text()).toBe("Cancel workflow")
```

Comment:
This comment lines can be removed. Instead check that the buttons are displayed but disabled.

Reply:
That won't work for the current implementation, the buttons aren't currently rendered, using `v-if="false"` - might be better just to have a test for no buttons here, which will fail and need to be modified when the buttons are added.